### PR TITLE
Set WSMAN_OPTION_UNENCRYPTED_MESSAGES when Basic auth is used over HTTP

### PR DESCRIPTION
**Pull request by dantraMSFT**
## PR Summary

This changes ensure WSMAN_OPTION_UNENCRYPTED_MESSAGES is set when using Basic auth over HTTP.  On Linux and macOS, not setting this option explicitly fails the connection request.

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:https://github.com/PowerShell/psl-omi-provider/issues/114
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [X] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs [hunk 1]
```diff
@@ -1526,9 +1526,12 @@ private void Initialize(Uri connectionUri, WSManConnectionInfo connectionInfo)
                 // config provider.
                 SetWSManSessionOption(WSManNativeApi.WSManSessionOption.WSMAN_OPTION_USE_SSL, 1);
             }
-            if (connectionInfo.NoEncryption)
+            if (connectionInfo.NoEncryption ||
```
Comment (PaulHigin) [inline src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs]: After thinking about this I am uncomfortable making PSRP messages non-encrypted by default for Basic auth.  Basic auth is not recommended but I am sure many customers use it, and may rely on PSRP messages being encrypted.

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs (hunk 1 above)
Comment (dantraMSFT) [reply src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs]: @PaulHigin: I limited the change to UNIX. PSRP/MI don't support encryption over HTTP like PSRP/WIndows/WSMan does.

**Comment by dantraMSFT** [comment]
FYI: The -workingdirectory test failure is being addressed with PR https://github.com/PowerShell/PowerShell/pull/6723

**Review by PaulHigin — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
I'm concerned about this being enabled by default on Unix.  On Windows, the user needs to enable `AllowUnencrypted` for both the client and server to work and is only intended for developer use.  Basic auth over unencrypted HTTP means the creds are in plain text over the wire.  Seems like we should make the user specify explicitly they are ok with this somehow.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]
This should be explicitly opt-in

**Comment by SteveL-MSFT** [comment]
@dantraMSFT not replace, but perhaps add that switch

**Comment by paulcallen** [comment]
encryption over HTTP is only supported with Kerberos, NTML and CredSSP (the last one not supported on Linux)

**Pull request closed without merging**